packages/devtools_app/lib/src/memory/memory_controller.dart

```diff
@@ -761,8 +911,16 @@ class MemoryController extends DisposableController
     snapshotByLibraryData ??= lastSnapshot?.librariesToList();
   }
 
-  void storeSnapshot(DateTime timestamp, HeapSnapshotGraph graph) {
-    snapshots.add(Snapshot(timestamp, this, graph));
+  void storeSnapshot(
+    DateTime timestamp,
+    HeapSnapshotGraph graph,
+    LibraryReference libraryRoot,
+  ) {
+    snapshots.value.add(Snapshot(timestamp, this, graph, libraryRoot));
+  }
+
+  void clearAllSnapshots() {
+    snapshots.value.clear();
```

Comment:
needs to be
`snapshots.value = []`
to notify given value notifier restrictions dealing with lists.

Reply:
Again not a ValueNotifier.